src/main/java/org/folio/entlinks/controller/delegate/AuthoritySourceFileServiceDelegate.java

```diff
@@ -72,13 +77,19 @@ public void patchAuthoritySourceFile(UUID id, AuthoritySourceFilePatchDto partia
     log.debug("patch:: Attempting to patch AuthoritySourceFile [id: {}, patchDto: {}]", id, partiallyModifiedDto);
     var existingEntity = service.getById(id);
     validateActionRightsForTenant(DomainEventType.UPDATE);
-    validatePatchRequest(partiallyModifiedDto, existingEntity);
+    var hasAuthorityReferences = anyAuthoritiesExistForSourceFile(existingEntity);
+    validatePatchRequest(partiallyModifiedDto, existingEntity, hasAuthorityReferences);
 
     var partialEntityUpdate = new AuthoritySourceFile(existingEntity);
     partialEntityUpdate = mapper.partialUpdate(partiallyModifiedDto, partialEntityUpdate);
     normalizeBaseUrl(partialEntityUpdate);
-    var patched = service.update(id, partialEntityUpdate);
+
+    var publishConsumer = publishRequired(hasAuthorityReferences, partiallyModifiedDto, existingEntity)
+      ? getUpdatePublishConsumer() : null;
+    var patched = service.update(id, partialEntityUpdate, publishConsumer);
     log.debug("patch:: Authority Source File partially updated: {}", patched);
+    ((ConsortiumAuthoritySourceFilePropagationService) propagationService)
```

Comment:
Just inject `ConsortiumAuthoritySourceFilePropagationService` instead of `ConsortiumPropagationService<AuthoritySourceFile>`

Reply:
Done